### PR TITLE
871 clarify channels behaviour and fix a markdown issue in API spec

### LINKED ISSUE #871: Clarify differences between App and User Channel behaviour (in docs for FDC3 versions 1.1, 1.2, 2.0 and next)
### Area of Issue
[x] API

### Issue Description: 
There is a key difference in behaviour between App Channels and System/User channels in FDC3 versions 1.1, 1.2, 2.0 in that, when you join a user channel (or add a context listener when already joined to the channel) your handler immediately receives the current context of the channel (filtered by the type of the listener). However, this is not the case when working with a channel through the Channel interface. In that case, you do NOT receive the current context but can retrieve it manually with `channel.getCurrentContext(/* optional context type name */)`.

This is currently (i.e. in 2.0) only stated in the overview:
2.0:
> When an app joins a User channel, or adds a context listener when already joined to a channel, it will automatically receive the current context for that channel.

https://fdc3.finos.org/docs/api/spec#joining-user-channels

1.2
> When an app joins a channel it will automatically recieve the current context for that channel.

https://fdc3.finos.org/docs/1.2/api/spec#joining-channels

and not in the function references for either the main Desktop Agent API or Channel interface. Hence, it is often missed by both Desktop Agent developers and application developers. Additional copy should be added to past and current documentation to clarify the behavior that should be expected.

### PR DESCRIPTION
**Pull request by kriswest**
resolves #871 

- Adds an additional clarifying note on user/app channel behaviour differences.
- Corrects a few spelling errors in API spec and Americanizes some spellings 🤢 
- Fixes a markdown rendering issue in the Desktop Agent specification I encountered while editing:

    image
    
    image

### CODE REVIEW

File: website/versioned_docs/version-2.0/api/ref/Channel.md [hunk 1]
```diff
@@ -6,9 +6,15 @@ original_id: Channel
...
 
-A channel can be either a "User" channel (retrieved with [`getUserChannels`](DesktopAgent#getuserchannels)) or a custom "App" channel (obtained through [`getOrCreateChannel`](DesktopAgent#getorcreatechannel)).
+A channel can be either a ["User" channel](../api/spec#joining-user-channels) (retrieved with [`getUserChannels`](DesktopAgent#getuserchannels)), a custom ["App" channel](../api/spec#app-channels) (obtained through [`getOrCreateChannel`](DesktopAgent#getorcreatechannel)) or a ["Private" channel](../api/spec#private-channels) (obtained via an intent result).
+
+:::note
+
+There are differences in behavior when you interact with a User channel via the `DesktopAgent` interface and the `Channel` interface. Specifically, when 'joining' a User channel or adding a context listener when already joined to a channel via the `DesktopAgent` interface, existing context (matching the type of the context listener) on the channel is received by the context listener immediately. Whereas, when a context listener is added via the Channel interface, context is not received automatically, but may be retrieved manually via the [`getCurrentContext()`](#getcurrentcontext) function.
```
Comment (greyseer256) [inline website/versioned_docs/version-2.0/api/ref/Channel.md:15]:
```suggestion
There are differences in behavior when you interact with a User channel via the `DesktopAgent` interface and the `Channel` interface. Specifically, when 'joining' a User channel or adding a context listener when already joined to a channel via the `DesktopAgent` interface, existing context (matching the type of the context listener) on the channel is received by the context listener immediately. Whereas, when a context listener is added via the `Channel` interface, context is not received automatically, but may be retrieved manually via the [`getCurrentContext()`](#getcurrentcontext) function.
```

**Review by greyseer256 — APPROVED** [review]
Looks good!

**Comment by greyseer256** [comment]
Actually, this PR doesn't include changes to the 1.2 docs, which is mentioned in the issue and I know we had discussed. Do we still want to include those updates? I'm happy to make those changes and update this branch.

**Comment by kriswest** [comment]
@greyseer256 If you're up for making the 1.2 changes, please go for it. No need to touch the /src/api/* file comments however, unless we want to release an npm module version with updated comments and a 1.2.x version number. The website's documentation is more accessible however.

**Review by mistryvinay — APPROVED** [review]
LGTM

**Comment by kriswest** [comment]
@greyseer256 on reflection, I'm not that worried about 1.2 docs. Still happy for the addition to go there too, but I think people will mostly read the current version of the documentation (and NPM module with comments).

**Comment by greyseer256** [comment]
@kriswest sorry it took me so long, I just filed a PR (into this branch) applying the same language to the 1.2 version of the docs:
https://github.com/finos/FDC3/pull/995

File: docs/api/spec.md [hunk 2]
```diff
@@ -525,7 +522,7 @@ const recommendedChannels = [
...
+While joining User channels (using fdc3.joinUserChannel) automates a lot of the channel behavior for an app, it has the limitation that an app can only be 'joined' to one channel at a time.  However, an app may instead retrieve a `Channel` Object via the [`fdc3.getOrCreateChannel`](ref/DesktopAgent#getorcreatechannel) API, or by raising an intent that returns a channel. The `Channel` object may then be used to listen to and broadcast on that channel directly using the [`Channel.addContextListener`](ref/Channel#addcontextlistener) and the [`Channel.broadcast`](ref/Channel#broadcast) APIs. This is especially useful for working with dynamic *App Channels*. FDC3 imposes no restriction on adding context listeners or broadcasting to multiple channels.
```
Comment (kriswest) [inline docs/api/spec.md:525]:
```suggestion
While joining User channels (using [`fdc3.joinUserChannel`](ref/DesktopAgent#joinuserchannel)) automates a lot of the channel behavior for an app, it has the limitation that an app can only be 'joined' to one channel at a time.  However, an app may instead retrieve a `Channel` Object via the [`fdc3.getOrCreateChannel`](ref/DesktopAgent#getorcreatechannel) API, or by raising an intent that returns a channel. The `Channel` object may then be used to listen to and broadcast on that channel directly using the [`Channel.addContextListener`](ref/Channel#addcontextlistener) and the [`Channel.broadcast`](ref/Channel#broadcast) APIs. This is especially useful for working with dynamic *App Channels*. FDC3 imposes no restriction on adding context listeners or broadcasting to multiple channels.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -17,6 +17,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 ### Fixed
 
+* Further clarified the difference between the behavior of User channels and other channel types on join/addContextListener. ([#971](https://github.com/finos/FDC3/pull/971))
```
Comment (kriswest) [inline CHANGELOG.md:20]:
```suggestion
* Further clarified the difference between the behavior of User channels and other channel types on joinUserChannel/addContextListener. ([#971](https://github.com/finos/FDC3/pull/971))
```

**Pull request merged**